doc/vhpidirect/examples/quickstart.rst

```diff
@@ -42,3 +42,40 @@ When the required functionality is not available in pre-built libraries, custom
 This example shows how to bind custom C functions in VHDL as either procedures or functions. Four cases are included: ``custom_procedure``, ``custom_procedure_withargs``, ``custom_function`` and ``custom_function_withargs``. In all cases, the parameters are defined as integers, in order to keep it simple. See :ref:`COSIM:VHPIDIRECT:Declarations` for further details.
 
 Since either C sources or pre-compiled ``.o`` objects can be added, in C/C++ projects of moderate complexity, it might be desirable to merge all the C sources in a single object before elaborating the design.
+
+:cosimtree:`logicvectors <vhpidirect/quickstart/logicvectors>`
+**************************************************************
+
+Commonly signals in VHDL are of a logic type or a vector thereof (``std_logic`` and ``std_logic_vector``), coming from the ``ieee.std_logic_1164.*`` package.
+Because logic values are more than high and low (``1`` and ``0``), the VHDL values of logic types (``std_logic`` and ``std_ulogic``) are enumrations:
+
+0. 'U' 
+1. 'X' 
+2. '0' 
+3. '1' 
+4. 'Z' 
+5. 'W' 
+6. 'L' 
+7. 'H' 
+8. '-' 
+
+Because number of enumeration values is less than 256, logic values are transported in 8 bit words (a ``char`` type in C).
+Logic vectors, of a bounded size, are easily created in C and passed to VHDL as ``char[]``. VHDL receives pointers as ``access`` types, in this case an access of a subtype of std_logic_vector.
```

Comment:
We should probably merge #10 first, and add a reference to that example here. #10 is focused on how to pass constrained arrays, only. This would be the next step, where enums are used (instead of ints or reals).

Reply:
:100:

Reply:
I also see the cPackageSimple (#7 ) as the first real addition to your examples, as it follows nicely after the `customc`.

Reply:
Yes. #7 goes first, because #10 uses a package. So: #7, #10, #6, #4, #5, #8, #3?

Reply:
There are 2 examples wrapped up in #5 (Sorry!), the first is accarrint (#5a), which is integer arrays with the bounded size. the second is #5b which uses the generic package. They both use packages... I would like to propose the following.
Make #5a without a package, so it can come before #10.
Make another version of #5b (called #5c) without a package so that it can follow #5a

Then I would have
1. #5a (No packages, int array passing)
2. #5c (No packages, int array, generic passing)
3. #6  (No packages, logic array passing) 
4. #7 (Packages)
5. #10 (Packages real array passing)
4. #4 (Packages, shared integer access)
6. #5b (Generic Packages, int array passing)
7. #8
8. #3

Reply:
Overall, I like the order. But I have been thinking about it and I think that we should probably create an explicit section about Arrays (vectors, multidimensional, constrained/bounded, unbounded, etc.). This is my proposal:

- Quick Start
  - random
  - math
  - customc
  - package (#7)
  - sharedvar (#4)
- Wrapping
  - basic
  - time
- Linking
  - bind
- Shared
  - shlib
  - dlopen
  - shghdl
- Arrays
  - intvector
    - Define from C only (#11)
      - without main
      - with main
    - Define from VHDL only (#12)
  - logicvector (#6)
  - matrices (#10)
  - genpkgs (#5)
- C interface
  - intro (#8)
  - demo (#3)

So, #4 is only dependent on #7, and I can work on finishing it.

Then, the chain would be:

- #11 & #12 (both merged into #11 first).
- #6 (this). It's ready. We just need to wait for #11 to be merged first, in order to update references accordingly.
- #10, is almost ready too.
- #5 will need some cleanup after all the previous steps are stabilized.
- #8 and #3, we'll see after all the others are done.

Reply:
>  an explicit section about Arrays 

Was feeling like it needed that too. I like your structure. I'll start moving the examples to an array/ subdirectory

I'll finish #12 and merge it into #11 then, High Priority 

Then I need to dive into #10 with a keen eye.